### PR TITLE
PE-66 Update deploy pipeline

### PR DESCRIPTION
**Pull request by ElfoLiNk**
#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -9,18 +9,41 @@ parameters:
...
+    FUNCTION_APP_NAME: $(UAT_FUNCTION_APP_NAME)
+    FUNCTION_APP_RESOURCE_GROUP_NAME: $(UAT_WEB_APP_RESOURCE_GROUP_NAME)
+    HEALTHCHECK_ENDPOINT: $(UAT_HEALTHCHECK_ENDPOINT)
+    HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME: $(UAT_HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME)
+    HEALTHCHECK_CONTAINER_VNET: $(UAT_HEALTHCHECK_CONTAINER_VNET)
+  ${{ if eq(parameters['environment'], 'prod') }}:
+    DEPLOY_TYPE: $[variables['PROD_DEPLOY_TYPE']]
+    AZURE_SUBSCRIPTION: $(PROD_AZURE_SUBSCRIPTION)
+    FUNCTION_APP_NAME: $(PROD_FUNCTION_APP_NAME)
+    FUNCTION_APP_RESOURCE_GROUP_NAME: $(PROD_WEB_APP_RESOURCE_GROUP_NAME)
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:40]:
```suggestion
    WEB_APP_NAME: $(PROD_WEB_APP_NAME)
    WEB_APP_RESOURCE_GROUP_NAME: $(PROD_WEB_APP_RESOURCE_GROUP_NAME)
```

Configs do not match with pipeline definition

File: .devops/deploy-pipelines.yml [hunk 2]
```diff
@@ -9,18 +9,41 @@ parameters:
...
-# the build and deploy.
+# variables setted on azdevops to avoid circular dependency warning
+# $(DEFAULT_BRANCH)
+# $(HEALTHCHECK_CONTAINER_SUBNET)
+variables:
+  ${{ if eq(parameters['environment'], 'uat') }}:
+    DEPLOY_TYPE: $[variables['UAT_DEPLOY_TYPE']]
+    AZURE_SUBSCRIPTION: $(UAT_AZURE_SUBSCRIPTION)
+    FUNCTION_APP_NAME: $(UAT_FUNCTION_APP_NAME)
+    FUNCTION_APP_RESOURCE_GROUP_NAME: $(UAT_WEB_APP_RESOURCE_GROUP_NAME)
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:32]:
```suggestion
    WEB_APP_NAME: $(UAT_WEB_APP_NAME)
    WEB_APP_RESOURCE_GROUP_NAME: $(UAT_WEB_APP_RESOURCE_GROUP_NAME)
```

File: .devops/deploy-pipelines.yml [hunk 3]
```diff
@@ -9,18 +9,41 @@ parameters:
...
+    AZURE_SUBSCRIPTION: $(PROD_AZURE_SUBSCRIPTION)
+    FUNCTION_APP_NAME: $(PROD_FUNCTION_APP_NAME)
+    FUNCTION_APP_RESOURCE_GROUP_NAME: $(PROD_WEB_APP_RESOURCE_GROUP_NAME)
+    HEALTHCHECK_ENDPOINT: $(PROD_HEALTHCHECK_ENDPOINT)
+    HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME: $(PROD_HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME)
+    HEALTHCHECK_CONTAINER_VNET: $(PROD_HEALTHCHECK_CONTAINER_VNET)
+
 pool:
-  vmImage: 'windows-2019'
+  vmImage: 'ubuntu-latest'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:46]:
```suggestion
  vmImage: 'windows-2019'
```
We use windows function-app please use windows image machine

File: .devops/deploy-pipelines.yml [hunk 4]
```diff
@@ -34,48 +57,47 @@ stages:
...
       - job: make_release
+        condition: eq(variables['Build.SourceBranch'], format('refs/heads/{0}', variables.DEFAULT_BRANCH))
         steps:
-        - ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/master') }}:
           - template: templates/node-job-setup/template.yaml@pagopaCommons
             parameters:
               persistCredentials: true
           - template: templates/node-github-release/template.yaml@pagopaCommons
             parameters:
               semver: '${{ parameters.RELEASE_SEMVER }}'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:82]:
```suggestion
              release_branch: '$(DEFAULT_BRANCH)'
              semver: '${{ parameters. DEFAULT_BRANCH }}'
```
not needed but better to specify

File: .devops/deploy-pipelines.yml [hunk 5]
```diff
@@ -116,49 +138,66 @@ stages:
...
+          - checkout: none
+          - download: current
+            artifact: Bundle
+
+          - task: AzureFunctionApp@1
             inputs:
-              azureSubscription: '$(PRODUCTION_AZURE_SUBSCRIPTION)'
-              resourceGroupName: '$(PRODUCTION_RESOURCE_GROUP_NAME)'
+              azureSubscription: '$(AZURE_SUBSCRIPTION)'
+              resourceGroupName: '$(FUNCTION_APP_RESOURCE_GROUP_NAME)'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:158]:
```suggestion
              resourceGroupName: '$(WEB_APP_RESOURCE_GROUP_NAME)'
```

File: .devops/deploy-pipelines.yml [hunk 6]
```diff
@@ -116,49 +138,66 @@ stages:
...
               appType: 'functionApp'
-              appName: '$(PRODUCTION_APP_NAME)'
+              appName: '$(FUNCTION_APP_NAME)'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:160]:
```suggestion
              appName: '$(WEB_APP_NAME)'
```

File: .devops/deploy-pipelines.yml [hunk 7]
```diff
@@ -116,49 +138,66 @@ stages:
...
-    jobs:
-      - job: 'do_healthcheck'       
-        steps:  
           - template: templates/rest-healthcheck/template.yaml@pagopaCommons
             parameters:
-              azureSubscription: '$(PRODUCTION_AZURE_SUBSCRIPTION)'
-              appName: '$(PRODUCTION_APP_NAME)'
-              endpoint: 'https://$(PRODUCTION_APP_NAME)-staging.azurewebsites.net/api/v1/info'
+              azureSubscription: '$(AZURE_SUBSCRIPTION)'
+              appName: '$(FUNCTION_APP_NAME)'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:170]:
```suggestion
              appName: '$(WEB_APP_NAME)'
              resourceGroupName: '$(WEB_APP_RESOURCE_GROUP_NAME)'
```

File: .devops/deploy-pipelines.yml [hunk 8]
```diff
@@ -116,49 +138,66 @@ stages:
...
-        steps:  
-          - task: AzureAppServiceManage@0   
+          - task: AzureAppServiceManage@0
             inputs:
-              azureSubscription: '$(PRODUCTION_AZURE_SUBSCRIPTION)'
-              resourceGroupName: '$(PRODUCTION_RESOURCE_GROUP_NAME)'
-              webAppName: '$(PRODUCTION_APP_NAME)'
-              sourceSlot: staging
+              azureSubscription: '$(AZURE_SUBSCRIPTION)'
+              resourceGroupName: '$(FUNCTION_APP_RESOURCE_GROUP_NAME)'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:180]:
```suggestion
              resourceGroupName: '$(WEB_APP_RESOURCE_GROUP_NAME)'
```

File: .devops/deploy-pipelines.yml [hunk 9]
```diff
@@ -116,49 +138,66 @@ stages:
...
+              webAppName: '$(FUNCTION_APP_NAME)'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:181]:
```suggestion
              appName: '$(WEB_APP_NAME)'
```

File: .devops/deploy-pipelines.yml [hunk 10]
```diff
@@ -116,49 +138,66 @@ stages:
...
+        condition: eq(variables.DEPLOY_TYPE, 'production_slot')
+        steps:
+          - checkout: none
+          - download: current
+            artifact: Bundle
+
+          - task: AzureWebApp@1
+            inputs:
+              azureSubscription: '$(AZURE_SUBSCRIPTION)'
+              resourceGroupName: '$(FUNCTION_APP_RESOURCE_GROUP_NAME)'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:196]:
```suggestion
              resourceGroupName: '$(WEB_APP_RESOURCE_GROUP_NAME)'
```

File: .devops/deploy-pipelines.yml [hunk 11]
```diff
@@ -116,49 +138,66 @@ stages:
...
+              appType: 'functionApp'
+              appName: '$(FUNCTION_APP_NAME)'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:198]:
```suggestion
              appName: '$(WEB_APP_NAME)'
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
check suggestions, maybe there was some errors

File: .devops/deploy-pipelines.yml [hunk 12]
```diff
@@ -116,49 +139,67 @@ stages:
...
+            displayName: 'Swap with production slot'
+
+      - job: 'do_deploy_production_slot'
+        condition: eq(variables.DEPLOY_TYPE, 'production_slot')
+        steps:
+          - checkout: none
+          - download: current
+            artifact: Bundle
+
+          - task: AzureWebApp@1
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:195]:
```suggestion
          - task: AzureFunctionApp@1
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
check suggestion

**Review by pasqualedevita — DISMISSED** [review]

**Comment by ElfoLiNk** [comment]
@AleDore UAT deployment done. Could you approve?

**Review by pasqualedevita — DISMISSED** [review]

**Review by AleDore — APPROVED** [review]

**Pull request merged**